src/pymor/models/interface.py

```diff
@@ -347,6 +419,53 @@ def output(self, mu=None, return_error_estimate=False, **kwargs):
         else:
             return data['output']
 
+    def solve_d_mu(self, parameter, index, mu=None, **kwargs):
+        """Solve for the partial derivative of the solution w.r.t. a parameter index
+
+        Parameters
+        ----------
+        parameter
+            parameter for which to compute the sensitivity
+        index
+            parameter index for which to compute the sensitivity
+        mu
+            |Parameter value| for which to solve
+
+        Returns
+        -------
+        The sensitivity of the solution as a |VectorArray|.
+        """
+        data = self.compute(
+            solution_d_mu=True,
+            mu=mu,
+            **kwargs
+        )
+        return data['solution_d_mu'][parameter][index]
+
+    def output_d_mu(self, mu=None, **kwargs):
+        """compute the gradient w.r.t. the parameter of the output functional
+
+        Parameters
+        ----------
+        mu
+            |Parameter value| for which to compute the gradient
+
+        Returns
+        -------
+        The gradient as a numpy array
+        """
+        data = self.compute(
+            output_d_mu=True,
+            mu=mu,
+            **kwargs
+        )
+        return data['output_d_mu']
+
+    @property
+    def dual(self):
+        """Instantiate the dual model which is used to solve for a dual solution of the |Model|."""
+        return NotImplemented
```

Comment:
I don't know if we need to add that to the interface. Maybe later when it is used by several Models?

Reply:
If we allow to pass a `dual_operator` for fixing the issue above, we could also already give a hint in the interface that we want to implement a dual model as a property, although no arbitrary definition is given.

Reply:
I just realized, we probably also instantiate `dual_operator` in `StationaryModel`, so then it would also be ok to remove `dual` from `Model`